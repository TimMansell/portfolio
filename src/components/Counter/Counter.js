import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

export const Counter = ({ begin, end }) => {
  const [counter, setCounter] = useState(begin);
  const [randomCountTo] = useState(
    Math.floor(Math.random() * (end - begin + 1) + begin) + 1
  );

  useEffect(() => {
    if (counter <= randomCountTo) {
      const timer = setInterval(() => {
        setCounter(counter + 1);
      }, 1);

      return () => clearInterval(timer);
    }
  });

  return <>{counter}+</>;
};

Counter.propTypes = {
  end: PropTypes.number.isRequired,
  begin: PropTypes.number.isRequired,
};

export default Counter;
