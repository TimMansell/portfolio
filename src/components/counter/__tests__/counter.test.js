import React from 'react';
import { shallow } from 'enzyme';

import Counter from '../counter';

describe("Counter", () => {
  it("should render my component", () => {
    const wrapper = shallow(<Counter />);
  });
});
