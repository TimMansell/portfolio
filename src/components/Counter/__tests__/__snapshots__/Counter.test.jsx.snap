// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Counter > should match snapshot 1`] = `"0"`;
